File: Npc/11004457.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11004457: Evian
/// </summary>
public class _11004457 : NpcScript {
    protected override int First() {
        return 10;
    }

    // Select 0:
    // $script:0109134107012662$
    // - Our first steps in a new world. I should be excited, but all I feel is nervous...
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (10, 0):
                // $script:0109134107012663$
                // - Our first steps in a new world. I should be excited, but all I feel is nervous...
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (10, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11004457.py
""" 11004457: Evian """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 10

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0109134107012662$
        # - Our first steps in a new world. I should be excited, but all I feel is nervous...
        return None # TODO

    def __10(self, pick: int) -> int:
        # $script:0109134107012663$
        # - Our first steps in a new world. I should be excited, but all I feel is nervous...
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (10, 0):
            return Option.CLOSE
        return Option.NONE
